src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs

```diff
@@ -119,7 +119,14 @@ public async Task StartAsync()
         /// <returns><c>true</c> if delay between next time we should ask for blocks is required; <c>false</c> otherwise.</returns>
         protected async Task<bool> SyncDepositsAsync()
         {
-            // First ensure that we the node is out of IBD.
+            // First ensure that the federation wallet is active.
+            if (!this.federationWalletManager.IsFederationWalletActive())
```

Comment:
I noticed in some logs that the MBSM is called whilst the node is still loading, which isn't correct behavior.

```
2021-07-20 13:08:13.8501 1] INFO: Stratis.Features.Collateral.CollateralChecker.InitializeAsync Initializing federation member XXXSV1x8yiAHAQESw58XcTg57aBVbNbUHz with amount 100000.00000000.
[2021-07-20 13:08:13.8767 1] INFO: Stratis.Features.Collateral.CollateralChecker.InitializeAsync Initializing federation member XX1gRPYpK2qiMvyjgHwV46R8Eaq94U3ZTN with amount 100000.00000000.
[2021-07-20 13:08:13.8792 1] INFO: Stratis.Features.Collateral.CollateralChecker.InitializeAsync Initializing federation member XXkQFV9Rc9WAkQe1SN6Ng75mtSLGFsKAV3 with amount 100000.00000000.
[2021-07-20 13:08:13.8792 1] INFO: Stratis.Features.Collateral.CollateralChecker.InitializeAsync Initializing federation member XY3HdNnGBWb5JkBchQDvYrkHGs38t6Qyq7 with amount 100000.00000000.
[2021-07-20 13:08:13.8792 1] INFO: Stratis.Features.Collateral.CollateralChecker.InitializeAsync Initializing federation member XCggPRCgSNRuhXF2xUE7v24UA92CCFaCGx with amount 100000.00000000.
[2021-07-20 13:08:13.8792 1] INFO: Stratis.Features.Collateral.CollateralChecker.InitializeAsync Initializing federation member XLjEnTqaroDqWdXeBt5cPP79pvYd48y1R7 with amount 100000.00000000.
[2021-07-20 13:08:13.8792 1] INFO: Stratis.Features.Collateral.CollateralChecker.InitializeAsync Initializing federation member XR1SRpgnbYdTMfMdyD9x1tUgucezGsnt7F with amount 100000.00000000.
[2021-07-20 13:08:14.0913 22] INFO: Stratis.Features.FederatedPeg.TargetChain.MaturedBlocksSyncManager.ProcessMatureBlockDepositsAsync Adding conversion fee distribution for transaction 'a1131ab1e82a8d2614f61305fcb3671e63464ded87977ea68a138d213e55d3e3' to deposit list.
[2021-07-20 13:08:14.1133 17] INFO: Stratis.Features.FederatedPeg.TargetChain.CrossChainTransferStore+<>c__DisplayClass41_1.<RecordLatestMatureDepositsAsync>b__5 1847 blocks received, containing a total of 18 deposits.
[2021-07-20 13:08:14.1133 17] INFO: Stratis.Features.FederatedPeg.TargetChain.CrossChainTransferStore+<>c__DisplayClass41_1.<RecordLatestMatureDepositsAsync>b__5 Block Range : 469303 to 471149.
[2021-07-20 13:08:14.1133 17] ERROR: Stratis.Features.FederatedPeg.TargetChain.CrossChainTransferStore+<>c__DisplayClass41_1.<RecordLatestMatureDepositsAsync>b__5 The store can't persist mature deposits while the federation is inactive.
[2021-07-20 13:08:14.1824 17] INFO: Stratis.Bitcoin.Utilities.AsyncLoop.StartAsync Notify starting.
[2021-07-20 13:08:14.1824 1] INFO: Stratis.Bitcoin.Features.MemoryPool.MempoolManager.LoadPoolAsync Loading Memory Pool.
[2021-07-20 13:08:14.1927 17] INFO: Stratis.Bitcoin.Utilities.AsyncLoop.StartAsync MemoryPool.RelayWorker starting in 10 seconds.
[2021-07-20 13:08:14.1927 1] INFO: Stratis.Bitcoin.Features.RPC.RPCFeature.InitializeAsync RPC Server is off based on configuration.
[2021-07-20 13:08:14.1927 1] INFO: Stratis.Bitcoin.Features.SmartContracts.SmartContractFeature.InitializeAsync Smart Contract Feature Injected.
[2021-07-20 13:08:14.1927 1] INFO: Stratis.Bitcoin.Features.ExternalApi.ExternalApiPoller.Initialize External API feature enabled, initializing periodic loops.
[2021-07-20 13:08:14.2072 1] INFO: Stratis.Bitcoin.Features.ExternalApi.ExternalApiPoller.Initialize Ethereum gas price tracking enabled.
[2021-07-20 13:08:14.2072 1] INFO: Stratis.Bitcoin.Features.ExternalApi.ExternalApiPoller.Initialize Price tracking for STRAX and ETH enabled.
[2021-07-20 13:08:14.2072 17] INFO: Stratis.Bitcoin.Utilities.AsyncLoop.StartAsync PeriodicCheckGasPrice starting in 1 seconds.
[2021-07-20 13:08:14.2072 18] INFO: Stratis.Bitcoin.Utilities.AsyncLoop.StartAsync PeriodicCheckPrice starting in 1 seconds.
[2021-07-20 13:08:14.2072 1] INFO: Stratis.Bitcoin.Features.Interop.InteropPoller.Initialize Interoperability enabled, initializing periodic loop.
[2021-07-20 13:08:14.2229 18] INFO: Stratis.Bitcoin.Utilities.AsyncLoop.StartAsync PeriodicCheckInterop starting in 1 seconds.
[2021-07-20 13:08:14.2229 17] INFO: Stratis.Bitcoin.Utilities.AsyncLoop.StartAsync PeriodicCheckConversionStore starting in 1 seconds.
[2021-07-20 13:08:14.2229 1] INFO: Stratis.Bitcoin.Features.Wallet.WalletManager.Start Wallet Manager starting...
```